src/LowercaseSentenceTokenizer.java

```diff
@@ -29,8 +30,32 @@ public class LowercaseSentenceTokenizer implements Tokenizer {
    * @return a list of tokens, where each token is a word or a period
    */
   public List<String> tokenize(Scanner scanner) {
-    // TODO: Implement this function to convert the scanner's input to a list of words and periods
-    return null;
+    List<String> tokens = new ArrayList<>();
+    StringBuilder stringInput = new StringBuilder();
+
+    while (scanner.hasNextLine()) {
+      //Reads the next line of text and adds to the stringInput object
+      stringInput.append(scanner.nextLine());
+    }
+
+    String stringLower = stringInput.toString().toLowerCase();
+
+    //Split by space and newlines
+    String[] splitString = stringLower.split("\\s+");
```

Comment:
This works well! However, `Scanner` by default splits on whitespace. Can you think of how you could use `scanner.next` instead of `scanner.nextLine` to simplify your code?